sdk/metric/sdk.go

```diff
@@ -336,39 +346,100 @@ func DefaultErrorHandler(err error) {
 // Labels returns a LabelSet corresponding to the arguments.  Passed
 // labels are de-duplicated, with last-value-wins semantics.
 func (m *SDK) Labels(kvs ...core.KeyValue) api.LabelSet {
-	// Note: This computes a canonical encoding of the labels to
-	// use as a map key.  It happens to use the encoding used by
-	// statsd for labels, allowing an optimization for statsd
-	// batchers.  This could be made configurable in the
-	// constructor, to support the same optimization for different
-	// batchers.
-
 	// Check for empty set.
 	if len(kvs) == 0 {
 		return &m.empty
 	}
 
-	ls := &labels{
-		meter:  m,
-		sorted: kvs,
+	ls := &labels{ // allocation
+		meter: m,
+		slice: kvs,
 	}
 
-	// Sort and de-duplicate.
-	sort.Stable(&ls.sorted)
+	// Sort and de-duplicate.  Note: this use of `ls.slice` avoids
+	// an allocation by using the address-able field rather than
+	// `kvs`.  Labels retains a copy of this slice, i.e., the
+	// initial allocation at the varargs call site.
+	//
+	// Note that `ls.slice` continues to refer to this memory,
+	// even though a new array is allocated for `ls.ordered`.  It
+	// is possible for the `slice` to refer to the same memory,
+	// although in the reflection code path of `computeOrdered` it
+	// costs an allocation to yield a slice through
+	// `(reflect.Value).Interface()`.
+	//
+	// TODO: It would be better overall if the export.Labels interface
```

Comment:
How about filing an issue about it when this PR gets merged?

Reply:
Will do.

Reply:
#539